package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+        if(!/(^\w+:)|(^\/)|(\.js$)/.test(value)) {
+          // Considered relative.
+          // Also works well if the value has ./ or ../
+          var id = this.id;
+          value = (id ? (id + "/") : "") + value;
+        }
+
+        return value;
+      },
+      //endregion
+
+      //region isAbstract property
+      // @type boolean
+      // -> boolean, Optional(false)
+
+      // Default value is for `Type` only.
+      // @see Type#constructor.
+      _isAbstract: true,
+
+      /**
+       * Gets or sets a value that indicates if this type is abstract.
+       *
+       * This attribute can only be set once, upon the type definition.
+       *
+       * @type {boolean}
+       * @default false
+       */
+      get isAbstract() {
+        return this._isAbstract;
+      },
+
+      set isAbstract(value) {
+        // nully is reset, which is false, so !! works well.
+        this._isAbstract = !!value;
+      },
+      //endregion
+
+      //region label property
+      // must have some non-null value to inherit
+      _label: "instance",
+      _labelSet: false,
+
+      /**
+       * Gets or sets the label of this type.
+       *
+       * When set to a non-{@link Nully} and non-{@link String} value,
+       * the value is first replaced by the result of calling its `toString` method.
+       *
+       * When set to an empty string or a _nully_ value, the attribute value is _reset_.
+       *
+       * When reset, the attribute assumes its _default value_
+       * (except on the top-root type, `Instance.type`, in which case it has no effect).
+       *
+       * The _default value_ is the _inherited value_.
+       *
+       * The _initial value_ of the attribute on the top-root type is `"instance"`.
+       *
+       * @type {!nonEmptyString}
+       */
+      get label() {
+        return this._label;
+      },
+
+      set label(value) {
+        value = nonEmptyString(value);
+        if(value === null) {
+          this._resetLabel();
+        } else {
+          this._labelSet = true;
+          this._label = value;
+        }
+      },
+
+      _resetLabel: function() {
+        this._labelSet = false;
+
+        if(this !== _type) {
+          delete this._label;
+        }
+      },
+      //endregion
+
+      //region description property
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized
+      // "" -> null conversion
+
+      _description: null, // set through implement bundle, below
+
+      /**
+       * Gets or sets the description of this type.
+       *
+       * Attempting to set to a non-string value type implicitly converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where this attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
```

Comment:
Suggest:

Setting this to `null` or to an empty string clears the attribute and sets it to `null`,